From 149cf6a4cecb2a165b550735fee3034d8c99b4bb Mon Sep 17 00:00:00 2001
From: Phil Burk <philburk@google.com>
Date: Mon, 25 Jun 2018 11:20:47 -0700
Subject: [PATCH 08/80] MediaExtractor: stop rendering when an error occurs

Bug: 68664359
Bug: 110435401
Test: Try to play a bogus OTA file. See bug for details.
Change-Id: If1322524fcfebc6c5f139288f044b0189da66c1b
---
 media/extractors/midi/MidiExtractor.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/media/extractors/midi/MidiExtractor.cpp b/media/extractors/midi/MidiExtractor.cpp
index 949fbe00d..a30b6f810 100644
--- a/media/extractors/midi/MidiExtractor.cpp
+++ b/media/extractors/midi/MidiExtractor.cpp
@@ -247,8 +247,9 @@ MediaBufferBase* MidiEngine::readBuffer() {
         EAS_I32 numRendered;
         EAS_RESULT result = EAS_Render(mEasData, p, mEasConfig->mixBufferSize, &numRendered);
         if (result != EAS_SUCCESS) {
-            ALOGE("EAS_Render returned %ld", result);
-            break;
+            ALOGE("EAS_Render() returned %ld, numBytesOutput = %d", result, numBytesOutput);
+            buffer->release();
+            return NULL; // Stop processing to prevent infinite loops.
         }
         p += numRendered * mEasConfig->numChannels;
         numBytesOutput += numRendered * mEasConfig->numChannels * sizeof(EAS_PCM);
-- 
2.17.1

